huaweicloud/services/dds/resource_huaweicloud_dds_recycle_policy.go

```diff
@@ -0,0 +1,133 @@
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"retention_period_in_days": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceDDSRecyclePolicyCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.DdsV3Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DDS client: %s ", err)
+	}
+
+	err = updateRecyclePolicy(client, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.IsNewResource() {
+		uuid, err := uuid.GenerateUUID()
+		if err != nil {
+			return diag.Errorf("unable to generate ID: %s", err)
+		}
+		d.SetId(uuid)
+	}
+
+	return resourceDDSRecyclePolicyRead(ctx, d, meta)
+}
+
+func updateRecyclePolicy(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	updateHttpUrl := "v3/{project_id}/instances/recycle-policy"
+	updatePath := client.Endpoint + updateHttpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"recycle_policy": map[string]interface{}{
+				"enabled":                  true,
+				"retention_period_in_days": d.Get("retention_period_in_days"),
+			},
+		},
+	}
+
+	_, err := client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return fmt.Errorf("error updating DDS recycle policy: %s", err)
+	}
+
+	return nil
+}
+
+func resourceDDSRecyclePolicyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.DdsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DDS client: %s ", err)
+	}
+
+	getHttpUrl := "v3/{project_id}/instances/recycle-policy"
+	getPath := client.Endpoint + getHttpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return diag.Errorf("error getting DDS recycle policy: %s", err)
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.Errorf("error flattening response: %s", err)
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("retention_period_in_days", utils.PathSearch("recycle_policy.retention_period_in_days", getRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceDDSRecyclePolicyDelete(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	errorMsg := "Deleting recycle policy resource is not supported. The resource is only removed from the state," +
```

Comment:
looks the default value of days is 7? you can use resetting the days to 7 as the delete func

Reply:
done